### PR TITLE
[WFCORE-5712] Update elytron-tool scripts to make use of jboss-modules

### PR DESCRIPTION
**Pull request by fjuma**
https://issues.redhat.com/browse/WFCORE-5712
https://issues.redhat.com/browse/WFCORE-5720
https://issues.redhat.com/browse/WFCORE-5730

Depends on https://github.com/wildfly/wildfly-core/pull/4878 (I've included those commits in this PR to also un-ignore `ElytronToolScriptTestCase`)

### CODE REVIEW

**Comment by fjuma** [comment]
@jamezp When you are back next week, would it be possible to get a review on the updates to the elytron-tool.[sh|bat|ps1] scripts? Thanks.

**Comment by fjuma** [comment]
I can't reproduce the `ElytronToolScriptTestCase` failures locally on Linux. Need to look closer at what's causing this to fail on CI.

**Comment by gaol** [comment]
I am not sure about the `ElytronToolScriptTestCase` failures, but the failure of `LayersTestCase.testServlet()` in full integration  should be fixed via https://github.com/wildfly/wildfly/pull/14970.

**Comment by jamezp** [comment]
@fjuma Is there a reason we remove then re-add the module? I assume it's for backwards compatibility, but will that work if the wrapper library is not provided?

**Comment by fjuma** [comment]
@jamezp `ELYTRON_TOOL_ADDONS` allows a user to optionally specify their own JAR that contains a custom credential store implementation. Before we updated the script to make use of jboss-modules, if a user defined `ELYTRON_TOOL_ADDONS`, it was just added to the classpath by the elytron-tool script.

After updating the script to make use of jboss-modules, this classpath approach could no longer work. So instead, if a user defines `ELYTRON_TOOL_ADDONS`, then the script will just create a new module that contains this JAR using the `module add` command. For the `module add` command to succeed, the module cannot already exist. That's why the script first removes the custom module if it happened to exist and then adds it.

File: core-feature-pack/common/src/main/resources/content/bin/elytron-tool.bat [hunk 1]
```diff
@@ -45,21 +49,29 @@ if "x%JAVA_HOME%" == "x" (
...
+set "DEPENDENCIES=java.logging,org.apache.commons.lang3,org.apache.commons.cli,org.apache.sshd,org.jboss.logging,org.jboss.logmanager,org.slf4j,org.wildfly.security.elytron-private,org.wildfly.common"
+set "MODULE_REMOVE_COMMAND=module remove --name=org.wildfly.security.elytron-tool-addons"
+set "MODULE_ADD_COMMAND=module add --name=org.wildfly.security.elytron-tool-addons --resources=%ELYTRON_TOOL_ADDONS% --dependencies=%DEPENDENCIES%"
+
 if not "x%ELYTRON_TOOL_ADDONS%" == "x" (
-   set ELYTRON_TOOL_SEP=;
-)
+  if exist "%JBOSS_MODULEPATH%\org\wildfly\security\elytron-tool-addons" (
 
+    call %JBOSS_CLI% --command=^"%MODULE_REMOVE_COMMAND%^" > nul
```
Comment (jamezp) [inline core-feature-pack/common/src/main/resources/content/bin/elytron-tool.bat:67]: This should be:
```
call "%JBOSS_CLI%" --command=^"%MODULE_REMOVE_COMMAND%^" > nul
```

File: core-feature-pack/common/src/main/resources/content/bin/elytron-tool.bat [hunk 2]
```diff
@@ -45,21 +49,29 @@ if "x%JAVA_HOME%" == "x" (
...
+  )
+  call %JBOSS_CLI% --command=^"%MODULE_ADD_COMMAND%^" > nul
```
Comment (jamezp) [inline core-feature-pack/common/src/main/resources/content/bin/elytron-tool.bat:69]: This should be:
```
call "%JBOSS_CLI%" --command=^"%MODULE_ADD_COMMAND%^" > nul
```

**Comment by fjuma** [comment]
@jamezp Thanks for the feedback! I've just merged a commit from @Ashpan that should address your comments on the .bat script.

**Review by jamezp — APPROVED** [review]
CI is still waiting to run, but the change itself looks okay to me. I did run the script tests on a Windows VM and Linux. They passed on both platforms.

**Comment by jamezp** [comment]
That galleon failure needs to be addressed. There is likely an issue on Windows too with the PowerShell script test. This can be fixed with a commit like https://github.com/wildfly/wildfly-core/commit/e6dd9280ae1401c3847a9b8fc45cc625ea0bd4cc.

**Comment by fjuma** [comment]
Thanks very much @jamezp! I've included your commit in this PR. 

The galleon failure should be addressed by https://github.com/wildfly/wildfly/pull/14970.

**Review by jamezp — APPROVED** [review]

**Comment by yersan** [comment]
@fjuma I did not realize it was going to conflict with another one merged. Could you rebase and resolve the conflict? thanks

**Comment by fjuma** [comment]
@yersan I've rebased and resolved the conflict. Thanks!

File: testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java [hunk 3]
```diff
@@ -49,6 +49,10 @@
         "org.wildfly.event.logger",
         // wildfly-elytron-http-stateful-basic
         "org.wildfly.security.http.sfbasic"
+        // wildfly-elytron-tool
+        "org.apache.commons.cli",
```
Comment (yersan) [inline testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java:53]: Hello @fjuma , missed comma "," after "org.wildfly.security.http.sfbasic"

Comment (fjuma) [reply testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java:53]: Sorry, fixed.

**Comment by yersan** [comment]
The `RemotingLegacySubsystemTestCase.testSubsystemWithConnectorPropertyChange` failing on "Linux JDK 11" Job is an intermittent error that it is unrelated to this PR, see [XNIO-385](https://issues.redhat.com/browse/XNIO-385)

**Comment by yersan** [comment]
Holding this PR until see how it integrates with https://github.com/wildfly/wildfly/pull/14988, at least at product branches there is a hard relationship between them. We can follow the integration here:

https://ci.wildfly.org/viewLog.html?buildTypeId=WF_WildFlyCoreIntegrationExperiments&buildId=285378

**Pull request merged**